### PR TITLE
CORE-5338: add FindWithNamedQuery.avsc

### PR DESCRIPTION
**Pull request by dickon**
CORE-5338: add a simple FindWithNamedQuery message for the persistence API

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "type": "record",
+  "name": "FindWithNamedQuery",
+  "doc": "Find entities which match named query with given parameters.  Replies with {@link EntityResponse}",
+  "namespace": "net.corda.data.persistence",
+  "fields": [
+    {
+      "name": "queryName",
+      "type": "string",
+      "doc": "the name of a named query defined somewhere in the virtual node"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc]:
```suggestion
      "doc": "The name of a named query defined in the CPK."
```

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    },
+    {
+      "name": "parameters",
+      "type": {
+        "type": "map",
+        "values": "bytes"
+      },
+      "doc": "parameters for the query"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc]:
```suggestion
      "doc": "Parameters for the query. Value can be a complex type and is an AMQP serialised payload."
```

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {
+      "name": "queryName",
+      "type": "string",
+      "doc": "the name of a named query defined somewhere in the virtual node"
+    },
+    {
+      "name": "parameters",
+      "type": {
+        "type": "map",
+        "values": "bytes"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc]: I think we said this should be `bytes` or `null` as null is a valid paramater in a JPA query.

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "type": "record",
+  "name": "FindWithNamedQuery",
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc]: I think it's worth adding the offset and limit parameters too. I know we have 2 JIRAs splitting it up, but adding them now will avoid another API change that then needs a version bump.

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      "type": {
+        "type": "map",
+        "values": "bytes"
+      },
+      "doc": "parameters for the query, with each value represented as a bytes array payload serialized in AMQP format"
+    },
+    {
+       "name": "offset",
+       "type" : "long",
+       "doc": "offset into the query results"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc]: Brought the language in line with the public API:

```suggestion
       "doc": "The index of the first result in the query to return."
```

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      "type": "string",
+      "doc": "the name of a named query defined somewhere in the virtual node"
+    },
+    {
+      "name": "parameters",
+      "type": {
+        "type": "map",
+        "values": "bytes"
+      },
+      "doc": "parameters for the query, with each value represented as a bytes array payload serialized in AMQP format"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc]:
```suggestion
      "doc": "Parameters for the query, with each value represented as a bytes array payload serialized in AMQP format."
```

File: data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    },
+    {
+       "name": "offset",
+       "type" : "long",
+       "doc": "offset into the query results"
+    },
+    {
+        "name": "limit",
+        "type": "long",
+        "doc": "limit number of query results; 0 for no limit"
```
Comment (driessamyn) [inline data/avro-schema/src/main/resources/avro/net/corda/data/persistence/FIndWithNamedQuery.avsc]: The public API says Int.Max is the default, so I don't think we should say 0 is a limit here. I think it's ok for it to be a mandatory property in our internal API and we'll take whatever is passed in. (So `0` would mean don't return anything).

Also brought the language in line with the public API.

```suggestion
        "doc": "Maximum number of results to return."
```

**Review by driessamyn — CHANGES_REQUESTED** [review]
Sorry, bit nitpicky about the documentation, but I think it's important style is consistent with other schemas a we may be generating documentation from them.

I realise that some others in this directory are also not consistent, but I will try to fix that up with a separate PR.

**Review by driessamyn — DISMISSED** [review]
LGTM

**Review by lankydan — DISMISSED** [review]

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**